Modern perl does not include '.' in its include path.
Need to patch the GNU make test.

diff -ru make-4.2.1~/Makefile.am make-4.2.1/Makefile.am
--- make-4.2.1~/Makefile.am	2016-05-22 13:16:06.000000000 +0000
+++ make-4.2.1/Makefile.am	2018-05-14 08:32:17.213552437 +0000
@@ -189,7 +189,7 @@
 		   done; fi ;; \
 	    esac; \
 	    echo "cd tests && $(PERL) ./run_make_tests.pl -srcdir $(abs_srcdir) -make ../make$(EXEEXT) $(MAKETESTFLAGS)"; \
-	    cd tests && $(PERL) ./run_make_tests.pl -srcdir '$(abs_srcdir)' -make '../make$(EXEEXT)' $(MAKETESTFLAGS); \
+	    cd tests && $(PERL) -I. ./run_make_tests.pl -srcdir '$(abs_srcdir)' -make '../make$(EXEEXT)' $(MAKETESTFLAGS); \
 	  else \
 	    echo "Can't find a working Perl ($(PERL)); the test suite requires Perl."; \
 	  fi; \
